VueJS/index.js

```diff
@@ -1,13 +1,35 @@
 import List from './components/List.js'
 import ListItem from './components/ListItem.js'
 
-new Vue({
+var myApp = new Vue({
   el: '#app',
+  methods: {
+	addTodo: function(){
+		this.todos.push(
+			{ id: this.getNextId(), message: '', }
+		);
+	},
+	saveTodo: function(todo){
+		this.todos[todo.index].message = todo.message;
+	},
+	deleteTodo: function(index){
+		this.todos.splice(index, 1);
+	},
+	getNextId: function(){
+	  return this.nextId++;
+	}
+  },
+  created: function(){
+	  if(this.nextId === null){
+		  this.nextId = this.todos.length+1;
+	  }
```

Comment:
I tried this but it caused errors when deleting an existing todo before creating a new one  because this introduced duplicate ids. A random id as you suggest is probably the best solution. Thanks for your feedback!

Reply:
This seems to work quite well: https://github.com/m-oehme/Systems-Development-and-Frameworks/pull/2#discussion_r339841486

But honestly, you rarely have to generate ids on the frontend. Most of the time, they come from the backend and get generated for you.